src/Pages/Shared/CollapsibleList/CollapsibleList.component.tsx

```diff
@@ -0,0 +1,131 @@
+/*
+ *   Copyright (c) 2024 Devtron Inc.
+ *   All rights reserved.
+
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+
+ *   http://www.apache.org/licenses/LICENSE-2.0
+
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+import React, { Fragment, useEffect, useState } from 'react'
+import { NavLink } from 'react-router-dom'
+import Tippy, { TippyProps } from '@tippyjs/react'
+
+import { ConditionalWrap } from '@devtron-labs/devtron-fe-common-lib'
+
+import { ReactComponent as ICExpand } from '../../../assets/icons/ic-expand.svg'
+
+import { CollapsibleListProps } from './CollapsibleList.types'
+import './CollapsibleList.scss'
+
+const renderWithTippy = (tippyProps: TippyProps) => (children: React.ReactElement) => (
+    <Tippy {...tippyProps}>
+        <div className="dc__align-self-start">{children}</div>
+    </Tippy>
+)
+
+export const CollapsibleList = ({ config, expandedIds }: CollapsibleListProps) => {
+    const [isExpanded, setIsExpanded] = useState<Record<string, boolean>>({})
+
+    useEffect(() => {
+        // Expand the collapsible list items whenever the `expandedIds` is modified.
+        setIsExpanded((prevState) => ({
+            ...prevState,
+            ...expandedIds.reduce((acc, id) => ({ ...acc, [id]: true }), {}),
+        }))
+    }, [expandedIds])
+
+    const handleCollapseBtnClick = (id: string) => () =>
+        setIsExpanded((prev) => ({
+            ...prev,
+            [id]: !prev[id],
+        }))
```

Comment:
How about we let the consumer component control this state altogether with change handler